### ISSUE TITLE
Expose distribution from `sample_latent` in `AutoContinuous`

### ISSUE THREAD

**Issue by fehiepsi** [uber/pyro#1232]
The current logic of all autoguide's method is:
```py
def sample_latent(self, *args, **kwargs):
    param = pyro.param(...)
    dist = dist.Distribution(param)
    return pyro.sample(name, dist, infer={"is_auxiliary": True})
```
It would be nice if we return `dist` in some method (e.g. `get_dist` / `get_posterior`) and refactor `sample_latent` into
```py
def sample_latent(self, *args, **kwargs):
    dist = self.get_dist(*args, **kwargs)
    return pyro.sample("_{}_latent".format(self.prefix), dist, infer={"is_auxiliary": True})
```

With the method `get_dist`, users can draw samples from this posterior or calculate other statistics.

**Comment by eb8680** [comment]
Seems like a nice enhancement, feel free to open a PR

**Comment by eb8680** [comment]
Resolved by #1240
